src/ch/imedias/rsccfx/model/connectionutils/RunRudp.java

```diff
@@ -76,8 +76,11 @@ public void run() {
         //Could be an issue in a p2p session behind NAT, works locally.
         //    rudpSocket = new ReliableSocket(model.getRemoteClientIpAddress().getHostAddress(),
         // model.getRemoteClientPort(), null, model.getIcePort());
+       // rudpSocket = new ReliableSocket(model.getRemoteClientIpAddress().getHostAddress(),
+         //   model.getRemoteClientPort());
+
         rudpSocket = new ReliableSocket(model.getRemoteClientIpAddress().getHostAddress(),
-            model.getRemoteClientPort());
+            model.getRemoteClientPort(), null, model.getIcePort());
```

Comment:
If null is handed over it allocates a port on all network cards. So the chance that we actually suceed having a connection over p2p is bigger

@param address   the remote address.
   * @param port      the remote port.
   * @param localAddr the local address the socket is bound to, or
   *                  <code>null</code> for the wildcard address.
   * @param localPort the local port the socket is bound to, or
   *                  <code>zero</code> for a system selected free port.

Currently we use a fixed port. I wanted to try handing over no port (using a different constructor) and expected it would allocate a random free one. This works locally but not over NAT...
One last thing to try out would be handing over null as localaddr and 0 as port.